pylint/checkers/typecheck.py

```diff
@@ -579,6 +582,72 @@ def _emit_no_member(
     return True
 
 
+def _get_all_attribute_assignments(
+    node: astroid.FunctionDef, name: Optional[str] = None
+) -> Set[str]:
+    attributes = set()
+    for child in node.nodes_of_class(astroid.Assign):
```

Comment:
There could be a false positive if an instance variable is annotated in the `__init__` method, for example:
```python
def __init__(unit):
    self.unit: str = unit
```

Reply:
Thanks for catching that, have pushed a fix :)

Reply:
Nice!